seed/seedwriter/writer.go

```diff
@@ -784,32 +763,23 @@ func (c byCompName) Len() int           { return len(c) }
 func (c byCompName) Less(i, j int) bool { return c[i].ComponentName < c[j].ComponentName }
 func (c byCompName) Swap(i, j int)      { c[i], c[j] = c[j], c[i] }
 
-func (w *Writer) assignLocalComponents(sn *SeedSnap) error {
-	// Add matching local components now that we know the snap name
-	for path, compInfo := range w.byPathLocalComps {
-		if compInfo.Component.SnapName != sn.SnapName() {
-			continue
-		}
-
+func (w *Writer) assignLocalComponents(sn *SeedSnap, seedComps map[string]*SeedComponent) error {
+	for _, seedComp := range seedComps {
 		// Check if the component is defined by the snap
-		compInSnap, ok := sn.Info.Components[compInfo.Component.ComponentName]
+		compInSnap, ok := sn.Info.Components[seedComp.ComponentName]
 		if !ok {
 			return fmt.Errorf("component %s is not defined by snap %s",
-				compInfo.Component.ComponentName, sn.SnapName())
+				seedComp.ComponentName, sn.SnapName())
 		}
 		// and if types match
-		if compInSnap.Type != compInfo.Type {
+		if compInSnap.Type != seedComp.Info.Type {
 			return fmt.Errorf("component %s has type %s while snap %s defines type %s for it",
-				compInfo.Component.ComponentName, compInfo.Type,
+				seedComp.ComponentName, seedComp.Info.Type,
```

Comment:
I think both of these branches are missing tests.

Reply:
There were tests in `image/image_test.go` but it is true that there were no tests in the `seedwriter` package, I've added them now.